### PR TITLE
python311Packages.sphinxawesome-theme: init at 5.1.4

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  buildPythonPackage,
+  fetchPypi,
+  lib,
+  poetry-core,
+  sphinx,
+  beautifulsoup4
+}:
+
+buildPythonPackage rec {
+  pname = "sphinxawesome_theme";
```
Comment (drupol) [inline pkgs/development/python-modules/sphinxawesome-theme/default.nix:11]:
```suggestion
  pname = "sphinxawesome-theme";
```

PEP503 normalization. Use a dedicated string for the src pname.

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-OwikuKJrPo4vNaud/9JToYYJePV6Kew8izYbr/qKTtQ=";
+  };
+
+  prePatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "^7.2,<7.3" ">=7.3"
+  '';
```
Comment (drupol) [inline pkgs/development/python-modules/sphinxawesome-theme/default.nix:23]: Don't use that, use `pythonRelaxDepsHook`, it has been made for that.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix (hunk 1 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix:11]: got it!

**Review by drupol — APPROVED** [review]
Thank you.

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix (hunk 1 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: this cause a `404` from `fetchPypi`, so i did
```nix
  pname = "sphinxawesome-theme";
  version = "5.1.4";
  pyproject = true;

  src = fetchPypi {
    inherit version;
    pname = "sphinxawesome_theme";
    hash = "sha256-OwikuKJrPo4vNaud/9JToYYJePV6Kew8izYbr/qKTtQ=";
  };

``

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix (hunk 2 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: I didn't knew that existed :o

Cant make it work though, may you help me on this one?

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: This is indeed how it's meant to be done, thank you.

**Comment by Sigmanificient** [comment]
@drupol `pythonRelaxDepsHook` doesn't seems to work:

```
/build/sphinxawesome_theme-5.1.4
Finished executing pypaBuildPhase
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for sphinxawesome_theme-5.1.4-py3-none-any.whl
  - sphinx<7.3,>=7.2 not satisfied by version 7.3.7
```
Cant make it work though, may you help me on this one?

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix (hunk 2 above)
Comment (drupol) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: Here's a very recent example: https://github.com/NixOS/nixpkgs/pull/315400/files#diff-83b3b91e3066d434e86c608d2f57d37c5cc5d806c433728b39e9de106f23ef50

Comment (drupol) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: Does this help?

Comment (Sigmanificient) [reply pkgs/development/python-modules/sphinxawesome-theme/default.nix]: perfect!

File: pkgs/development/python-modules/sphinxawesome-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  version = "5.1.4";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "sphinxawesome_theme";
+    hash = "sha256-OwikuKJrPo4vNaud/9JToYYJePV6Kew8izYbr/qKTtQ=";
+  };
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
```
Comment (drupol) [inline pkgs/development/python-modules/sphinxawesome-theme/default.nix:22]: I think you can safely merge `nativeBuildInputs` and `build-system`. Hope @mweinelt is OK with that :)

**Pull request merged**